pinot-core/src/main/java/com/linkedin/pinot/core/data/manager/realtime/LLRealtimeSegmentDataManager.java

```diff
@@ -840,6 +844,20 @@ public LLRealtimeSegmentDataManager(RealtimeSegmentZKMetadata segmentZKMetadata,
     // Create field extractor
     _fieldExtractor = FieldExtractorFactory.getPlainFieldExtractor(schema);
     makeConsumerWrapper();
+
+    // Partitioners
+    _column2PartitionersMap = indexingConfig.getPartitioners();
+    if (_column2PartitionersMap != null && !_column2PartitionersMap.isEmpty()) {
+      try {
+        _nPartitions = _consumerWrapper.getPartitionCount(_kafkaTopic, /*maxWaitTimeMs=*/5000L);
+        _segmentPartitionConfig = createPartitionConfig();
```

Comment:
Do you need _segmentPartitionConfig as a member variable ? If it is not used anywhere else then make it local